cumulus/parachains/runtimes/collectives/collectives-westend/src/secretary/mod.rs

```diff
@@ -0,0 +1,196 @@
+// Copyright (C) Parity Technologies (UK) Ltd.
+// This file is part of Cumulus.
+
+// Cumulus is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Cumulus is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Cumulus.  If not, see <http://www.gnu.org/licenses/>.
+
+//! The Polkadot Secretary Collective.
+//!
+//! The module defines the following on-chain functionality of the Secretary Program:
+//!
+//! - Managed set of program members, where every member is at the same rank
+//! (via [SecretaryCollective](pallet_ranked_collective)).
+//! - Referendum functionality for the program members to propose, vote on, and execute
+//! proposals on behalf of the program (via [SecretaryReferenda](pallet_referenda)).
+//! - Promotion and demotion periods, register of members' activity, and rank based salaries
+//! (via [SecretaryCore](pallet_core_fellowship)).
+//! - Members' salaries (via [SecretarySalary](pallet_salary), requiring a member to be
+//! imported or inducted into [SecretaryCore](pallet_core_fellowship)).
+pub mod origins;
+mod tracks;
+
+use super::*;
+use crate::xcm_config::{FellowshipAdminBodyId, LocationToAccountId, WndAssetHub};
+use frame_support::traits::{tokens::GetSalary, EitherOf, EitherOfDiverse, MapSuccess};
+use frame_system::EnsureRootWithSuccess;
+pub use origins::pallet_origins as pallet_secretary_origins;
+use origins::pallet_origins::EnsureSecretary;
+use sp_core::ConstU128;
+use sp_runtime::traits::{ConstU16, ConvertToValue, Identity, Replace, ReplaceWithDefault};
+use xcm::prelude::*;
+use xcm_builder::{AliasesIntoAccountId32, PayOverXcm};
+
+/// The Secretary members' ranks.
+pub mod ranks {
+	use pallet_ranked_collective::Rank;
+
+	#[allow(dead_code)]
+	pub const SECRETARY_CANDIDATE: Rank = 0;
+	pub const SECRETARY: Rank = 1;
+}
+
+/// Origins of:
+/// - Root;
+/// - FellowshipAdmin (i.e. token holder referendum);
+/// - Plurality vote from Fellows can promote, demote, remove and approve rank retention
+/// of members of the Secretary Collective (rank `2`).
+type ApproveOrigin = EitherOf<
+	EnsureRootWithSuccess<AccountId, ConstU16<65535>>,
+	EitherOf<
+		MapSuccess<
+			EnsureXcm<IsVoiceOfBody<GovernanceLocation, FellowshipAdminBodyId>>,
+			Replace<ConstU16<2>>,
+		>,
+		MapSuccess<Fellows, Replace<ConstU16<2>>>,
```

Comment:
I do not see a reason to limit with 2, and might be easily forgotten to be updated